code-challenges.test.js

```diff
@@ -35,22 +87,94 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+// a describe method that lists the name of the function OR naming of the particular test.
 
-// b) Create the function that makes the test pass.
+describe("oddFilter", () => {
+    var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+    var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it("takes in an array and returns a new array of only odd numbers sorted from least to greatest.", () => {
 
+        //an expect method, nested within the test block, calling on the oddFilter() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+        expect(oddFilter(fullArr1)).toEqual([-9, 7, 9, 199])
+        expect(oddFilter(fullArr2)).toEqual([-823, 7, 23])
+    })
+})
+// -- Got the red --  
 
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// b) Create the function that makes the test pass.
+// create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+function oddFilter(array) {
+    // declare the an empty arrays
+    var oddNumArr = []
+    //filter out the array, only sort the odd numbers in array
+    oddNumArr = array.filter(number => number % 2 !== 0 && (typeof number) === "number")
+    //then sort the odd numbers from least to greatest
+    let leastToGreatest = oddNumArr.sort((a, b) => a - b);
+    //return the values
+    return leastToGreatest
+}
```

Comment:
Your logic looks good. If you are creating a variable just to return it, you can always shorten your code and just return the code logic. For example `return oddNumArr.sort((a, b) => a - b)`